static/main.js

```diff
@@ -6,20 +6,28 @@ const ctx = canvas.getContext("2d");
 canvas.height = 500;
 canvas.width = 500;
 
+// variable for initial board state
+var board_initial;
+
 // Socket.io
-var socket = io('http://localhost:8080')
+var socket = io('http://localhost:8080/canvas')
 socket.on('connect', function () {
 	console.log(socket.id);
 });
 
 socket.on('broadcast-board', function (imagedata) {
+	// turn JSON into imageData
+	imagedata = JSON.parse(imagedata);
+	array = new Uint8ClampedArray(imagedata.data);
 	console.log(imagedata);
-	ctx.putImageData(imageData, 0, 0);
+	// create new ImageData & update board
+	imagedata = new ImageData(array,imagedata.width,imagedata.height);
+	ctx.putImageData(imagedata, 0, 0);
+	board_initial = ctx.getImageData(0, 0, canvas.width, canvas.height);
```

Comment:
Let's grab the initial board state as close to when we are actually making the change as possible.

Reply:
Put 'board_initial = ctx.getImageData(0, 0, canvas.width, canvas.height);' in startPos() function to grab the most recent state of the board